### ISSUE TITLE
PATH order is inconsistent with zsh terminal

### ISSUE THREAD

**Issue by pptheberge** [wavetermdev/waveterm#368]
The order of the path entries doesn't seem to be respected. Here's the  content of a simplified .zprofile file

`export PATH=/test/path:$PATH
`

In a regular terminal, the path is set to:

`path=( /test/path /usr/local/bin /System/Cryptexes/App/usr/bin /usr/bin /bin /usr/sbin /sbin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin )
`

But with waveterm, the path is set to:

`path=( /usr/local/bin /System/Cryptexes/App/usr/bin /usr/bin /bin /usr/sbin /sbin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin /test/path )
`
Notice that the path element that should be inserted at the beginning of the path is added at the end instead.

This is causing issues with path overrides.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version 14.2.1

**Comment by sawka** [comment]
Thanks for submitting!  I just tried this out locally and can reproduce this behavior.  This is definitely a P0.  I'll work to get a fix for this ASAP.

**Comment by sawka** [comment]
Okay, I think I understand what's going on here.  It is an interaction between apple's path setup code and the (at least for me) unexpected way that zsh handles `typeset` and uniquing arrays.

So in /etc/zprofile apple runs this command: ```eval `/usr/libexec/path_helper -s` ```.  that actually sets up your path using what's in /etc/paths and /etc/paths.d/.  Now the way Wave restores your path is using a typeset command similar to `typeset -xUaT PATH path=( /test/path ... )`.  I debugged this an indeed "/test/path" *is* the first entry of the restored path.  What I didn't expect was that zsh's uniquing code doesn't respect the order of the typeset.  It actually respects the order of the initial PATH variable!  So all of the system paths come first, and then any new paths follow -- which is why you're seeing "/test/path" at the end -- because it follows all of the apple generated paths.

tl;dr I now understand what's going on so I can fix it.

**Comment by sawka** [comment]
https://gist.github.com/Linerre/f11ad4a6a934dcf01ee8415c9457e7b2

**Comment by pptheberge** [comment]
I just tried 0.7.0. The simplified test case works fine now but the real-life scenario still fails. We're using nvm to manage node versions and the path is still different. This is the regular terminal path

`path=( /Users/userdir/.nvm/versions/node/v21.6.2/bin /opt/local/bin /opt/local/sbin /usr/local/bin /System/Cryptexes/App/usr/bin /usr/bin /bin /usr/sbin /sbin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin )
`

But with waveterm we get this.

`path=( /usr/local/bin /System/Cryptexes/App/usr/bin /usr/bin /bin /usr/sbin /sbin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin /Users/userdir/.nvm/versions/node/v21.6.2/bin /opt/local/bin /opt/local/sbin )
`

Note that this is a real user setup. I'll try to create a simple reproducible test case.

**Comment by pptheberge** [comment]
It looks like I didn't understand how workspaces and tabs are supposed to behave.  Now that I realize that I have to run reset in each tab to reinitialize it and I can't just rely on just restaring waveterm, everything works. Thank you for the fix.
